code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -495,6 +506,35 @@
 		OpenFire(A)
 	..()
 
+/**
+  * Proc that handles the charge impact of the charging mob.
+  * * hit_atom: the target of the charge.
+  */
+/mob/living/simple_animal/hostile/throw_impact(atom/hit_atom)
+	if(SEND_SIGNAL(src, COMSIG_HOSTILE_POST_CHARGE) & !(COMPONENT_HOSTILE_POSTCHARGE_IMPACT))
+		return ..()
```

Comment:
This feels slightly dirty, but I'd love to be told this is valid.

Reply:
The correct syntax, if I get what you are trying to do, would be this:
`if(!(SEND_SIGNAL(src, COMSIG_HOSTILE_POST_CHARGE) & COMPONENT_HOSTILE_POSTCHARGE_IMPACT))`
Or maybe this:
`if(SEND_SIGNAL(src, COMSIG_HOSTILE_POST_CHARGE) & (ALL & ~COMPONENT_HOSTILE_POSTCHARGE_IMPACT))`

Reply:
Shouldn't all this be in the charger component.  This structure doesn't make much sense, you have very specific signal here that won't be used for anything else.
Any reason it can't be implemented as something like:
throw_impact > Sends on impact signal > Charger Component blocks default action, does it stuff, ideally calling some on impact signal other stuff can subscribe to